### PR TITLE
Add Debian 12

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: lib/puppet_metadata/operatingsystem.rb [hunk 1]
```diff
@@ -32,6 +32,7 @@ class OperatingSystem
       # https://wiki.debian.org/DebianReleases
       'Debian' => {
         # TODO: EOL is standard support, not the extended life cycle
+        '12' => nil, # '~2026',
         '11' => nil, # '~2024',
         '10' => nil, # '~2022',
```
Comment (evgeni) [inline lib/puppet_metadata/operatingsystem.rb:37]: technically, 10 is EOL by now. should we add the date now?

**Comment by bastelfreak** [comment]
In the past we waited with adding a new OS until puppet provides AIO packages for the platform. I don't mind adding Debian 12 already but then we need to ensure that we use the distro packages instead.

File: lib/puppet_metadata/operatingsystem.rb (hunk 1 above)
Comment (bastelfreak) [reply lib/puppet_metadata/operatingsystem.rb:37]: yes, we should do that in another PR (actually it was 2022-09-10 and we missed it) :D
https://endoflife.date/debian

**Comment by evgeni** [comment]
> In the past we waited with adding a new OS until puppet provides AIO packages for the platform. I don't mind adding Debian 12 already but then we need to ensure that we use the distro packages instead.

They still… Okay…

How do I tell it to use distro packages, is that even in this repo? Do I really want to know?

**Comment by ekohl** [comment]
It is in this repository: https://github.com/voxpupuli/puppet_metadata/commit/0dcad5388994a21805a3eaa5446bf61efc716ed0 is how it was done for Arch Linux.

It was in my wish list to support this at some point. We probably need some special casing here:

https://github.com/voxpupuli/puppet_metadata/blob/5ffbf7905b5b9ab1be5cf39d94291ba7b0f7a409/lib/puppet_metadata/github_actions.rb#L63-L67

We probably need something for Fedora too there.

**Comment by ekohl** [comment]
> We probably need something for Fedora too there.

https://github.com/voxpupuli/puppet_metadata/pull/104 does that. And I've taken the liberty of taking your PR and rebase it on top of it, doing the same for Debian 12.

**Comment by ekohl** [comment]
Testing it out in https://github.com/voxpupuli/puppet-example/pull/37.

File: lib/puppet_metadata/operatingsystem.rb [hunk 2]
```diff
@@ -206,6 +207,33 @@ def supported_releases(operatingsystem)
...
+      #
+      # Only tracks releases without an AIO build, since that's preferred.
+      #
+      # @param [String] operatingsystem
+      #   The operating system name
+      # @param [String] release
+      #   The operating system release version
+      #
+      # @return [Optional[Integer]] The Puppet major version, if any
+      def os_release_puppet_version(operatingsystem, release)
```
Comment (bastelfreak) [inline lib/puppet_metadata/operatingsystem.rb:221]: todo for me: add arch linux here.

**Review by bastelfreak — APPROVED** [review]

File: lib/puppet_metadata/operatingsystem.rb (hunk 2 above)
Comment (ekohl) [reply lib/puppet_metadata/operatingsystem.rb:221]: Since it's rolling I'm not sure if that's desirable.

**Review by bastelfreak — APPROVED** [review]

**Comment by ekohl** [comment]
Dependencies to merge & release this:
* https://github.com/voxpupuli/voxpupuli-acceptance/pull/70 & release as gem
* https://github.com/voxpupuli/beaker_puppet_helpers/pull/37 & release as gem

**Pull request merged**